src/mongocxx/test/client_helpers.hh

```diff
@@ -210,9 +210,37 @@ auto size(Container c) -> decltype(std::distance(std::begin(c), std::end(c))) {
 //
 // @return Whether sessions are supported by the client's topology.
 //
-bool server_has_sessions(const client& conn);
-
-bool should_run_client_side_encryption_test(void);
+bool server_has_sessions_impl(const client& conn);
+
+#define SERVER_HAS_SESSIONS_OR_SKIP(conn)                       \
+    if (!mongocxx::test_util::server_has_sessions_impl(conn)) { \
+        SKIP("server does not support session");                \
+    }                                                           \
+    ((void)0)
```

Comment:
Minor: Recommend an `else` here so that the macro expands to a single `if` statement.

Reply:
Actually: Does this need to be a macro? Does the `SKIP` need to expand within the body of the test case?

Reply:
> Does this need to be a macro? Does the `SKIP` need to expand within the body of the test case?

Yes, in order for source location information (filename and line number) to be unique in the skip reason message.